core/src/main/scala/org/apache/spark/storage/BlockManager.scala

```diff
@@ -221,6 +221,178 @@ private[spark] class BlockManager(
     new BlockManager.RemoteBlockDownloadFileManager(this)
   private val maxRemoteBlockToMem = conf.get(config.MAX_REMOTE_BLOCK_SIZE_FETCH_TO_MEM)
 
+  /**
+   * @param blockSize the decrypted size of the block
+   */
+  private abstract class BlockStoreUpdater[T](
+      blockSize: Long,
+      blockId: BlockId,
+      level: StorageLevel,
+      classTag: ClassTag[T],
+      tellMaster: Boolean,
+      keepReadLock: Boolean) {
+
+    protected def byteBuffer: ChunkedByteBuffer
+
+    protected def saveToDiskStore(): Unit
+
+    private def saveToMemoryStore(): Boolean = {
+      if (level.deserialized) {
+        val values =
+          serializerManager.dataDeserializeStream(blockId, byteBuffer.toInputStream())(classTag)
```

Comment:
If the goal here is to save memory, you could also do something here.

e.g., in the temp file case, instead of reading the data into a `ChunkedByteBuffer` first, your could return an input stream directly.